src/SubspaceClustering.jl

```diff
@@ -20,32 +20,30 @@ include("algorithms/kss.jl")
 
 # Utility functions
 """
-    randsubspace([rng=default_rng()], [T=Float64], D, d)
+    randsubspace([rng=default_rng()], [T=Float64], D, d) where T<:Number
 
-Generate a random `d`-dimensional subspace of `ℝᴰ`
-and return a basis matrix with element type `T<:AbstractFloat`.
+Generate a random `d`-dimensional subspace of `ℝᴰ` (if T<:AbstractFloat) or `ℂᴰ`
+(if T<:Complex{<:AbstractFloat}) and return a `D×d` orthonormal basis matrix with element type `T`.
 
 See also [`randsubspace!`](@ref)
 """
 randsubspace(rng::AbstractRNG, ::Type{T}, D::Integer, d::Integer) where {T<:AbstractFloat} =
     randsubspace!(rng, Array{T}(undef, D, d))
-randsubspace(::Type{T}, D::Integer, d::Integer) where {T<:AbstractFloat} =
+randsubspace(rng::AbstractRNG, ::Type{T}, D::Integer, d::Integer) where {T<:Complex{<:AbstractFloat}} =
+    randsubspace!(rng, Array{T}(undef, D, d))
```

Comment:
Rather than add this method that essentially duplicates the one above, let's instead just modify the one above by replacing `AbstractFloat` with `Union{AbstractFloat,Complex{<:AbstractFloat}}`. Ditto for the `where {T<:Number}` clause in the method below.

Read about `Union` types here: https://docs.julialang.org/en/v1/manual/types/#Type-Unions

And see an example of how it's used for SVD here:
https://github.com/JuliaLang/LinearAlgebra.jl/blob/c723c590f3184e170e0679e0f02b69dc5d0f1270/src/svd.jl#L182-L185